core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -159,24 +172,34 @@ case class ExecInfo(
     }
   }
 
-  def getUnsupportedExecSummaryRecord(execId: Long): Seq[UnsupportedExecSummary] = {
-    val unsupportedReason = getUnsupportedReason
-    val res =
-      ArrayBuffer(UnsupportedExecSummary(sqlID, execId, exec, opType,
-        unsupportedReason, getOpAction))
+  def getUnsupportedExecSummaryRecord(
+      execId: Long,
+      unSupportedOpsReasons: Map[String, String] = Map.empty): Seq[UnsupportedExecSummary] = {
+
+    // Get the custom reason if it exists
+    val execCustomReason = getCustomReason(exec, unSupportedOpsReasons)
+    val execUnsupportedReason = determineUnsupportedReason(execCustomReason, getUnsupportedReason)
+
+    // Initialize the result with the exec summary
+    val res = ArrayBuffer(UnsupportedExecSummary(sqlID, execId, exec, opType,
+      execUnsupportedReason, getOpAction, isExpression = false, execCustomReason))
+
     // TODO: Should we iterate on exec children?
-    // add the unsupported expressions to the results
+    // add the unsupported expressions to the results, if there are any custom reasons add them
+    // to the result appropriately
     if (unsupportedExprs.nonEmpty) {
-      // unsupported expression will depend on what we learned from the exec itself
-      val exprReason = unsupportedReason match {
+      val exprKnownReason = execUnsupportedReason match {
         case UnsupportedReasons.CONTAINS_UDF => UnsupportedReasons.IS_UDF
         case UnsupportedReasons.CONTAINS_DATASET => UnsupportedReasons.IS_DATASET
         case UnsupportedReasons.UNSUPPORTED_IO_FORMAT => UnsupportedReasons.UNSUPPORTED_IO_FORMAT
         case _ => UnsupportedReasons.IS_UNSUPPORTED
       }
+
```

Comment:
CUSTOM_REASON check is not required here since it is assigning reasons for expressions. In the subsequent lines, we check per expression if it has any custom reason and assign it accordingly.